mal_scrapper.py

```diff
@@ -0,0 +1,209 @@
+        scrapped = True
+
+    # If we find the anime, let's get the alternative titles
+    if name in saved_data:
+        anime = saved_data[name]
+
+        if 'alternative_titles' in anime.keys(): # Index in list, we use it
+            return anime['alternative_titles']
+
+        if not scrapped: # Index not in list but no scrapping done... Local data may not be updated so we do it now
+            return get_alternative_titles(tags, True)
+        
+    return []
+
+
+def get_artists(tags, force_scrap=False):
+    """
+    Find and save the artists of an anime theme by scrapping MyAnimeList.
+    The scrapped data will be cached to reduce later parsing times.
+
+    tags is a dictionnary generated by anime_parse.parse_file_name (will be edited with the found artists)
+    """
+
+    # We don't care about IS
+    if tags['link_type'] not in ['OP', 'ED']:
+        return []
+
+    # json_file_path = os.path.dirname(os.path.realpath(__file__)) + '/mal_scrapper.json'
+
+    # We first load the local data
+    with File(json_file_path, 'r') as json_file:
+        saved_data = json.load(json_file)
+
+    name = tags['title_work']
+    if tags['subtitle_work'] != '':
+        name = name + ' ~ ' + tags['subtitle_work']
+
+    # If we don't find the anime, let's scrap MAL
+    scrapped = False
+    if name not in saved_data or force_scrap: # We check if we don't already have the anime scrapped
+        scrap_anime(name, saved_data)
+        scrapped = True
+
+    # If we find the anime, let's get the artists
+    if name in saved_data:
+        anime = saved_data[name]
+        index = str(tags['link_nb'])
+        if index == '0': # If the index either 0, it will still be stored as 1
+            index = '1'
+
+        if index in anime[tags['link_type']]: # Index in list, we use it
+            return anime[tags['link_type']][index]
+
+        if not scrapped: # Index not in list but no scrapping done... Local data may not be updated so we do it now
+            return get_artists(tags, True)
+
+        # Index not in list despite scrapping... Can't do anything else here
+        print('\nWARNING MAL_Scrapper: ' + name + ' ' + tags['link_type'] + str(tags['link_nb']) + ' not found')
+        
+    return []
+
+
+def scrap_anime(name, data):
+    """
+    This function does the actual scrapping and saves it in data.
+    """
+
+    # First steap: finding the page of the anime, using the search engine
+    response = requests.get('https://myanimelist.net/anime.php?q="' + name + '"')
+    if not response.ok:
+        print('\nWARNING MAL_Scrapper: can\'t connect to MyAnimeList')
+        return
+
+    found = {}
+
+    kept_results_max = 10 # Max number of results kept
+
+    parsed = bs.BeautifulSoup(response.text, features='html.parser')
+    animes_list = parsed.find('div', {'class': 'js-block-list'}).table.findAll('tr')
+
+    for i in range(1, kept_results_max + 1): # We won't keep more than kept_results_max entries
```

Comment:
Beware that you never use the first element of the list (of index 0). If this is voluntary, you should add a comment.

A more Pythonic way to iterate over a portion of a list:

```python
for anime_row in animes_list[:kept_results_max]:
    pass
```